node-data/src/message.rs

```diff
@@ -1091,7 +1120,7 @@ pub mod payload {
         inventory: Inv,
 
         /// (requester) Address to which the resource is sent back, if found
-        requester_addr: SocketAddr,
+        requester_addr: Option<SocketAddr>,
```

Comment:
Why should this be an Option?

Reply:
Agree.
We should not use `GetResource msg` here but another internal message between Consensus and Chain that should be converted into a flood_request.  I decided to start with `GetResource`  to speed up testing.